src/AcmTable/AcmTable.tsx

```diff
@@ -605,22 +616,54 @@ export function AcmTable<T>(props: AcmTableProps<T>) {
         [filtered, rows, keyFn]
     )
 
-    const actions = rowActions.map((rowAction) => {
-        return {
-            title: rowAction.title,
-            onClick: (_event: React.MouseEvent, rowId: number, rowData: IRowData) => {
-                if (groupFn || addSubRows) {
-                    const tableItem =
-                        rowData.props?.key && sorted.find((tableItem) => tableItem.key === rowData.props.key)
-                    if (tableItem) {
-                        rowAction.click(tableItem.item)
+    // Function to parse provided actions from AcmTable IAcmRowAction --> Patternfly Table IAction
+    const parseRowAction = (rowActions: IAcmRowAction<T>[]) => {
+        const actions: IAction[] = []
+        rowActions.forEach((action) => {
+            // Add separator if specified
+            if (action.addSeparator) {
+                actions.push({
+                    isSeparator: true,
+                })
+            }
+            // Add row action
+            actions.push({
+                title: action.title,
+                onClick: (_event: React.MouseEvent, rowId: number, rowData: IRowData) => {
+                    if (groupFn || addSubRows) {
+                        const tableItem =
+                            rowData.props?.key && sorted.find((tableItem) => tableItem.key === rowData.props.key)
```

Comment:
```suggestion
                            rowData.props?.key && paged.find((tableItem) => tableItem.key === rowData.props.key)
```

`paged` will be a shorter list to search in